### PR TITLE
Fix password reset with null pointer exception

### PR DESCRIPTION
**Pull request by aeng**
fix null pointer exception when request for password reset.

### CODE REVIEW

File: zanata-war/src/main/java/org/zanata/service/impl/UserAccountServiceImpl.java [hunk 1]
```diff
@@ -58,24 +58,22 @@
...
         HAccountResetPasswordKey key =
-                (HAccountResetPasswordKey) session
-                        .createCriteria(HAccountResetPasswordKey.class)
-                        .add(Restrictions.eq("account", account))
-                        .uniqueResult();
-        if (key != null) {
-            session.delete(key);
-            session.flush();
-        }
+                accountResetPasswordKeyDAO.findByAccount(account.getId());
```
Comment (huangp) [inline zanata-war/src/main/java/org/zanata/service/impl/UserAccountServiceImpl.java]: It used to check whether the key is null. Is now safe to ignore the check?

Comment (aeng) [inline zanata-war/src/main/java/org/zanata/service/impl/UserAccountServiceImpl.java]: Good one. it should check for null.

File: zanata-war/src/main/java/org/zanata/action/PasswordResetRequestAction.java [hunk 2]
```diff
@@ -70,17 +70,19 @@
 
 
     public String requestReset() {
-        if(getAccount() == null) {
+        HAccountResetPasswordKey key =
+            userAccountServiceImpl.requestPasswordReset(getAccount());
+
+        if(getAccount() == null || key == null) {
```
Comment (seanf) [inline zanata-war/src/main/java/org/zanata/action/PasswordResetRequestAction.java]: You're using `getAccount()` (above) before checking if it's null.

File: zanata-war/src/main/java/org/zanata/action/PasswordResetRequestAction.java [hunk 3]
```diff
@@ -112,10 +114,10 @@ public boolean isAccountWaitingForActivation() {
         if (account == null) {
             return false;
         }
-        if (account.getAccountActivationKey() == null) {
-            return false;
+        if (account.getAccountActivationKey() != null) {
+            return true;
         }
-        return true;
+        return false;
```
Comment (seanf) [inline zanata-war/src/main/java/org/zanata/action/PasswordResetRequestAction.java]: How about `return account.getAccountActivationKey() != null` ?

File: zanata-war/src/main/java/org/zanata/action/PasswordResetRequestAction.java (hunk 2 above)
Comment (aeng) [inline zanata-war/src/main/java/org/zanata/action/PasswordResetRequestAction.java]: Good one.

File: zanata-war/src/main/java/org/zanata/service/impl/UserAccountServiceImpl.java [hunk 4]
```diff
@@ -58,24 +58,24 @@
...
-        // TODO This should be done in a DAO
         HAccountResetPasswordKey key =
-                (HAccountResetPasswordKey) session
-                        .createCriteria(HAccountResetPasswordKey.class)
-                        .add(Restrictions.eq("account", account))
-                        .uniqueResult();
-        if (key != null) {
-            session.delete(key);
-            session.flush();
+                accountResetPasswordKeyDAO.findByAccount(account.getId());
```
Comment (seanf) [inline zanata-war/src/main/java/org/zanata/service/impl/UserAccountServiceImpl.java]: +1 on the dao extraction.

**Comment by huangp** [comment]
:+1:

**Comment by djansen-redhat** [comment]
:white_check_mark: Tested.

**Comment by aeng** [comment]
Merged into integration/master. https://github.com/zanata/zanata-server/commit/3b0d55a8d4113479be77d0e9c667e9c4ef017066

**Pull request closed without merging**